### PR TITLE
feature(core): allow implementers to override the action done when a message is failed to be consumed #216

### PR DESCRIPTION
**Pull request by ahmadshabib**
…

## :loudspeaker: Type of change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
This implementation gives the ability to perform different actions when a message consumption fails, by providing a default implementation for the fail cases that can be overridden by the implementer

## :bulb: Motivation and Context
#216 

## :green_heart: How did you test it?
Locally due to the lack of a testing suite

## :pencil: Checklist

- [x] I reviewed the submitted code
- [ ] I added tests to verify changes
- [ ] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: src/main/kotlin/com/katanox/tabour/integration/sqs/core/consumer/SqsEventHandler.kt [hunk 1]
```diff
@@ -44,4 +46,8 @@ class SqsEventHandler(
             }
         }
     }
+
+    fun onFailure(throwable: Throwable, message: Any) {
+        failureAction
```
Comment (Mantas-Gav) [inline src/main/kotlin/com/katanox/tabour/integration/sqs/core/consumer/SqsEventHandler.kt:51]: just to be sure: you don't need to invoke it as a function?

**Review by Mantas-Gav — COMMENTED** [review]
Looks nice. Have you tried overriding the onFailure to see whether it really works?

File: src/main/kotlin/com/katanox/tabour/integration/sqs/core/consumer/SqsEventHandler.kt (hunk 1 above)
Comment (ahmadshabib) [reply src/main/kotlin/com/katanox/tabour/integration/sqs/core/consumer/SqsEventHandler.kt]: figure it that this is not needed

**Comment by ahmadshabib** [comment]
> Looks nice. Have you tried overriding the onFailure to see whether it really works?

image

**Review by Mantas-Gav — APPROVED** [review]

**Pull request merged**